### PR TITLE
[Python]: fix makefile and cmake for native python interpreter support

### PR DESCRIPTION
**Pull request by eddy16112**
Fix the compilation for native python interpreter

### CODE REVIEW

File: python/flexflow/core/__init__.py [hunk 1]
```diff
@@ -33,11 +33,20 @@
...
 
   # check if use native python interpreter
-  if flexflow_python_interpreter() == 'native':  
-    from .flexflow_pybind11_internal import begin_flexflow_task, finish_flexflow_task
+  if flexflow_python_interpreter() == 'native': 
     print("Using native python")
-    begin_flexflow_task(sys.argv)
-    atexit.register(finish_flexflow_task)
+    if flexflow_python_binding() == 'pybind11':
+      from .flexflow_pybind11_internal import begin_flexflow_task, finish_flexflow_task
```
Comment (lockshaw) [inline python/flexflow/core/__init__.py:39]: Aren't we getting rid of pybind support?

**Review by lockshaw — APPROVED** [review]
I don't see any glaring problems, though I'll admit that I'm not super familiar with this part of the code (tagging in @williamberman who has recently been touching the build system a bunch in case he's interested).

File: python/flexflow/core/__init__.py (hunk 1 above)
Comment (eddy16112) [reply python/flexflow/core/__init__.py:39]: Yes, but that will be in another PR, so I will just keep the pybind in this one.

**Comment by eddy16112** [comment]
@gabrieleoliaro The PR is ready, could you please take a look at it, especially the changes to the cmake?

**Comment by jiazhihao** [comment]
Great! I have enabled auto-merge --- the PR will be merged once it passes all checks.

**Pull request merged**